wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -529,6 +531,47 @@ createHdWallet newRoot defaultHdAccountId defaultHdAddress utxoByAccount =
                Nothing ->
                    Map.insert defaultHdAccountId (mempty, [defaultAddr]) m
 
+-- | Create an HdWallet with HdRoot
+--
+-- NOTE: We allow an initial set of accounts with associated addresses and
+-- balances /ONLY/ for testing purpose. Normally this should be empty; see
+-- 'createHdWallet'/'createWalletHdRnd' in "Cardano.Wallet.Kernel.Wallets".
+--
+-- INVARIANT: Creating a new wallet always come with a fresh HdAccount and
+-- a fresh 'HdAddress' attached to it, so we have to pass these two extra
+-- piece of into to the update function. We do @not@ build these inside the
+-- update function because derivation requires an 'EncryptedSecretKey' and
+-- definitely we do not want it to show up in our acid-state logs.
```

Comment:
This comment is wrong and copy-pasted from above I believe. We do not create any address in the HdWallet. 
This invariant was primarily a sugar for Daedalus and other clients, so I understand why we do not preserve it for the HdWallet; 
@denisshevchenko, any extra inputs maybe?